File: PRAK403-2310817210021-RAKHMAD FAUZHAN RAMADHAN.py
def generate_crossing_numbers(num1, num2):
    result = []

    if num1 > num2:
        for i in range(num1, num2 - 1, -1):
            result.append((i, num1 + num2 - i))
    elif num2 > num1:
        for i in range(num1, num2 + 1):
            result.append((i, num1 + num2 - i))

    return result

def main():
    input_str = input("Masukkan dua angka dipisahkan oleh spasi: ")
    numbers = [int(x) for x in input_str.split()]

    crossing_numbers = generate_crossing_numbers(numbers[0], numbers[1])

    output_str = " - ".join(["{} {}".format(x[0], x[1]) for x in crossing_numbers])
    print(output_str)

if __name__ == "__main__":
    main()

File: PRAK403-2310817210021-RAKHMAD FAUZHAN RAMADHAN.c
#include <stdio.h>

void generate_crossing_numbers(int num1, int num2) {
    if (num1 > num2) {
        for (int i = num1; i >= num2; i--) {
            printf("%d %d", i, num1 + num2 - i);
            if (i > num2) {
                printf(" - ");
            }
        }
    } else if (num2 > num1) {
        for (int i = num1; i <= num2; i++) {
            printf("%d %d", i, num1 + num2 - i);
            if (i < num2) {
                printf(" - ");
            }
        }
    }
}

int main() {
    int num1, num2;

    printf("Masukkan dua angka dipisahkan oleh spasi: ");
    scanf("%d %d", &num1, &num2);

    generate_crossing_numbers(num1, num2);

    return 0;
}
